src/v2/Apps/Conversation/Utils/groupMessages.tsx

```diff
@@ -0,0 +1,42 @@
+import { Conversation_conversation } from "v2/__generated__/Conversation_conversation.graphql"
+import { DateTime } from "luxon"
+import { fromToday } from "../Components/TimeSince"
+
+type Message = Conversation_conversation["messagesConnection"]["edges"][number]["node"]
+/**
+ * Combines messages into groups of messages sent by the same party and
+ * separated out into different groups if sent across multiple days
+ * @param messages Messages in the conversation
+ */
+export const groupMessages = (messages: Message[]): Message[][] => {
```

Comment:
**praise:** Moving this (and conversation items out) are solid refactors!

This is a pretty complex part of conversations anyway so moving it out definitely will help us think about it in isolation and (hopefully) test it better.